packages/plugin-senpi-rewards/src/actions/claimRewardAction.ts

```diff
@@ -0,0 +1,127 @@
+import {
+    type Action,
+    type IAgentRuntime,
+    type Memory,
+    type HandlerCallback,
+    type State,
+    elizaLogger,
+    type ActionExample,
+} from "@moxie-protocol/core";
+import { MoxieWalletClient } from "@moxie-protocol/moxie-agent-lib/src/wallet";
+import { encodeFunctionData} from "viem";
+
+import { getRewardBalance } from "../utils/balance";
+
+export const claimRewardsAction: Action = {
+    name: "CLAIM_REWARDS",
+    similes: [
+        "CLAIM_REWARDS",
+        "CLAIM_REWARDS_ON_SENPI",
+        "SEND_REWARDS_TO_SENPI_WALLET",
+    ],
+    description: "Send, payout or claim your earned rewards from Senpi to your wallet",
+    suppressInitialMessage: true,
+    validate: async () => true,
+    handler: async (
+        runtime: IAgentRuntime,
+        message: Memory,
+        state: State,
+        _options: { [key: string]: unknown },
+        callback: HandlerCallback
+    ) => {
+        try {
+            elizaLogger.log("Starting CLAIM_REWARDS handler...");
+
+            const { address } = state.agentWallet as MoxieWalletClient;
+
+            const balance = await getRewardBalance(address as `0x${string}`);
+
+            if (!balance || balance === 0) {
+                await callback?.({
+                    text: "You have no rewards to claim.",
+                    action: "CLAIM_REWARDS",
+                });
+                return;
+            }
+
+
+            console.log("---- balance", balance);
```

Comment:
use `elizaLogger` instead + add `traceId` for any logs

Might help w/ debugging in the future if needed